### PR TITLE
Issue/#56 crud json diagram config

### PR DESCRIPTION
**Pull request by korzol**
Hello @ceckoslab 
Kindly review the code

### CODE REVIEW

File: templates/widgets/register_form.html.twig [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<!-- Modal -->
```
Comment (ceckoslab) [inline templates/widgets/register_form.html.twig]: Please rename this file to edit.thiw.html

File: src/Controller/WidgetsController.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Entity\Widgets;
+use App\Repository\WidgetsRepository;
```
Comment (ceckoslab) [inline src/Controller/WidgetsController.php]: App\Repository\WidgetsRepository is missing and I don't find it on the file system.

```
In RegisterControllerArgumentLocatorsPass.php line 173:
Cannot determine controller argument for "App\Controller\WidgetsController::index()": the $widgetsRepository argument is type-hinted with the non-existent class or interface: "App\Repository\WidgetsRepository".
```

File: src/Controller/WidgetsController.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
+<?php
```
Comment (ceckoslab) [inline src/Controller/WidgetsController.php]: Please use the already existing WidgetController.php

**Comment by korzol** [comment]
Hello @ceckoslab 
Kindly review the latest updates

**Comment by ceckoslab** [comment]
Hello @korzol 

Did we implement a view action for viewing diagrams?

**Comment by korzol** [comment]
No, we don't. That's what I wonder about. How would we implement the view. Should I implement in onto dashboard page?

**Comment by ceckoslab** [comment]
@korzol let's do it in a view action. For example we will render a diagram when we visit e.g. widget/view/5

**Comment by korzol** [comment]
Hello @ceckoslab 
Kindly review updates

File: templates/widgets/view.html.twig [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    </div>
+</div> #}
+
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+       // alert("Ready!");
+        $(function(){
+        $.ajax('/widget/generate_diagram',{
+            method: 'post',
+            data: JSON.parse('{{ item.widget|raw }}'),
```
Comment (ceckoslab) [inline templates/widgets/view.html.twig:33]: The best will be if we define a JavaScript variable called widgetJson = item.widget|raw and we pass it to AJAX data. By doing this we don't need to have JSON.parse()

This is an example where it fails for me:

```
{
                title: 'Device Distribution',
                global: {
                    presentation: {
                        'render_type': 'distribution'
                    },
                    data_requirements: {
                        period: {
                            type: 'moving',
                            start: '30',
                            end: 'now',
                        }
                    }
                },
                segments: {
                    1: {
                        presentation: {
                            name: 'Desktop',
                            color: '#1F77B4'
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 2,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    },
                    2: {
                        presentation: {
                            name: 'Tablet',
                            color: '#ff6023'
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 3,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    },
                    3: {
                        presentation: {
                            name: 'Mobile',
                            color: '#2CA02C',
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 1,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    },
                    4: {
                        presentation: {
                            name: 'Bot',
                            color: '#000000'
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 4,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    }
                }
            }
```

**Comment by korzol** [comment]
Hello @ceckoslab 
Kindly review the code

**Pull request merged**